src/tools/interopserver/InteropServer.h

```diff
@@ -196,7 +253,12 @@ struct HttpServer {
         ) {
         if (Event->Type == QUIC_LISTENER_EVENT_NEW_CONNECTION) {
             Event->NEW_CONNECTION.SecurityConfig = SecurityConfig;
-            new HttpConnection(Event->NEW_CONNECTION.Connection);
+            if (Event->NEW_CONNECTION.Info->NegotiatedAlpnLength >= 6 &&
+                !memcmp(Event->NEW_CONNECTION.Info->NegotiatedAlpn, "siduck", 6)) {
```

Comment:
Can you not use strcmp here? Are they not null terminated?

Reply:
Nope. ALPN is **not** null terminated. That's why the length is also provided.